main.go

```diff
@@ -474,8 +486,22 @@ func secureNetwork(testNetwork *network.Network, decelerationFactor float64) {
 }
 
 func startSecurityWorker(peer *network.Peer, pace time.Duration) {
-	for range time.Tick(pace) {
-		sendMessage(peer)
+	if pace == time.Duration(0) {
+		return
+	}
+	ticker := time.NewTicker(pace)
+
+	for {
+		select {
+		case <-ticker.C:
+			if config.IMIF == "poisson" {
+				pace = time.Duration(float64(time.Second) * rand.ExpFloat64() / float64(pace))
+				if pace > 0 {
```

Comment:
Is it always > 0? ([ExpFloat64 Ref](https://pkg.go.dev/math/rand#ExpFloat64)) If so we don't have to check it.

Reply:
it can be 0, but Ticker needs to have duration > 0, that's why I added this check

Reply:
So maybe the GoLang doc is wrong? (`ExpFloat64 returns an exponentially distributed float64 in the range (0, +math.MaxFloat64]`)

Reply:
Rather I was trying some strange parameters and `pace` was < 1 and rounded to 0 when converting to duration